src/protocol/message_layer/StreamMessage.js

```diff
@@ -0,0 +1,52 @@
+import InvalidJsonError from '../../errors/InvalidJsonError'
+
+export default class StreamMessage {
+    constructor(version, ttl, contentType, content) {
+        if (new.target === StreamMessage) {
+            throw new TypeError('StreamMessage is abstract.')
+        }
+        this.version = version
+        this.ttl = ttl
+        this.contentType = contentType
+        this.content = content
+    }
+
+    getSerializedContent() {
+        if (typeof this.content === 'string') {
+            return this.content
+        } else if (this.contentType === StreamMessage.CONTENT_TYPES.JSON && typeof this.content === 'object') {
+            return JSON.stringify(this.content)
+        } else if (this.contentType === StreamMessage.CONTENT_TYPES.JSON) {
+            throw new Error('Stream payloads can only be objects!')
+        } else {
+            throw new Error(`Unsupported content type: ${this.contentType}`)
+        }
+    }
+
+    getParsedContent() {
```

Comment:
Should this just be done in constructor? One less method to worry about and failing early being the benefits. Now this is sort of a "lazy" method in a way, but I think this is always eventually called anyways?